src/postgres/src/backend/commands/createas.c

```diff
@@ -626,8 +626,9 @@ intorel_receive(TupleTableSlot *slot, DestReceiver *self)
 
 	if (IsYBRelation(myState->rel))
 	{
+		slot->tts_tupleDescriptor=RelationGetDescr(myState->rel);
```

Comment:
ExecSetSlotDescriptor should be used.
However, it is a bad idea to modify passed in  slot. 
Why do we think that slot does not have a descriptor already?

Reply:
I haven't used ExecSetSlotDescriptor as this clears the tuple from the slot. 
slot should have descriptor otherwise it won't work in slot_getattr as that retrieves tupleDesc from slot->tts_tupleDescriptor. But, I have done a check

RelationGetDescr(rel)!=slot->tts_tupleDescriptor -> this is true as it has different tdtypeid and tdrefcount. So, I tried to update it.

Based on your comments i have done a recheck and I hope we aren't using tdtypeid and tdrefcount anywhere.